pkg/block/index.go

```diff
@@ -245,11 +250,25 @@ func GatherIndexHealthStats(logger log.Logger, fn string, minTime, maxTime int64
 	}
 	stats.MetricLabelValuesCount = int64(len(lvals))
 
+	// As of version two all series entries are 16 byte padded. All references
+	// we get have to account for that to get the correct offset.
+	offsetMultiplier := 1
+	version := r.Version()
+	if version >= 2 {
+		offsetMultiplier = 16
+	}
+
 	// Per series.
+	var prevId storage.SeriesRef
```

Comment:
Is it impossible for a series ref to be 0? :thinking:

Reply:
We usually use a ref of 0 when appending to new series in tsdb, so I assume it's a reserved value: https://github.com/prometheus/prometheus/blob/394da05dd407f07b592b097ee6a7b4a54d9d9967/storage/interface.go#L220-L228